arpeggio/__init__.py

```diff
@@ -779,19 +779,27 @@ class RegExMatch(Match):
             It will be used to create regular expression using re.compile.
         ignore_case(bool): If case insensitive match is needed.
             Default is None to support propagation from global parser setting.
+        multiline(bool): allow regex to works on multiple lines (re.DOTALL flag)
         str_repr(str): A string that is used to represent this regex.
+        re_flags: flags parameter for re.compile if neither ignore_case
+            or multiple are set.
 
     '''
     def __init__(self, to_match, rule_name='', root=False, ignore_case=None,
-                 str_repr=None):
+                 multiline=False, str_repr=None, re_flags=re.MULTILINE):
         super(RegExMatch, self).__init__(rule_name, root)
         self.to_match_regex = to_match
         self.ignore_case = ignore_case
+        self.multiline = multiline
+        implicit_flag = any((self.ignore_case, self.multiline, to_match is None))
```

Comment:
Explicit `None` test should be made as the user might set some of the flags to `False`.
`to_match` is never `None`!
I think something like this:
```python
implicit_flag = self.ignore_case is not None or self.multiline is not None
```

Reply:
That seems more logical: `implicit_flag = (self.ignore_case or 0) | (self.multiline or 0)`.

Reply:
> `to_mach` is never `None` !
So what's the point of:
> have `flags` param that if not `None` is used in regex
?

Reply:
> That seems more logical: `implicit_flag = (self.ignore_case or 0) | (self.multiline or 0)`

These expressions are not equivalent. I though that `implicit_flag` should be `True` only if some of the param/flags is given (e.g. not None). Your expression will be `True` (1 actually) if some of the param is `True`, which is not what we are testing, right? 

> So what's the point of:

>> have `flags` param that if not `None` is used in regex

Sorry, maybe I wasn't clear enough. In the initial post I suggested to use flags param only if it is given, e.g. if its value (the value of `flags` param not `to_match`) is not `None` which is default. In that case additional paras (`ignore_case` and `multiline`) would be ignored.

Another solution would be to merge all params by applying `ignore_case` and `multiline` over given `re_flags` (which is by default `re.MULTILINE` to keep current behaviour) anyway.